docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,307 @@
+
+Vanilla adds an additional helper method to further streamline the process of checking these conditions: `Enchantment#applyEffects()`. This method takes a `List<ConditionalEffect<T>>`, evaluates the conditions, and runs a `Consumer<T>` on each `T` contained by a `ConditionalEffect` whose condition was met. Since many vanilla enchantment effect components are defined as a `List<ConditionalEffect<?>>`, these can be directly plugged into the helper method like so:
+
+```java
+// `enchant` is an Enchantment instance.
+// `lootContext` is a LootContext instance.
+Enchantment.applyEffects(
+    enchant.getEffects(EnchantmentEffectComponents.KNOCKBACK), // Or whichever other List<ConditionalEffect<T>> you want
+    lootContext,
+    (effectData) -> // Use the effectData (in this example, an EnchantmentValueEffect) however you want.
+);
+```
+
+Registering a custom `ConditionalEffect`-wrapped enchantment effect component type can be done as follows:
+
+```java
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ConditionalEffect<ExampleData>>> EXAMPLE_CONDITIONAL_EFFECT =
+    ENCHANTMENT_COMPONENT_TYPES.register("example_conditional",
+        () -> DataComponentType.ConditionalEffect<ExampleData>builder()
+            .persistent(ConditionalEffect.codec(Unit.CODEC, LootContextParamSets.EMPTY))
+            .build());
+```
+The parameters to `ConditionalEffect.codec` are the codec for the generic `ConditionalEffect<T>`, followed by some `LootContextParamSets` entry.
+
+### Using Enchantment Effect Components
+
+Here is a full example using vanilla helper methods to work with a custom enchantment effect component.
+
+```java
+// Define an example data-bearing record.
+public record Increment(int value) {
+    public static final Codec<BoundEntity> CODEC = RecordCodecBuilder.create(instance ->
+            instance.group(
+                    Codec.INT.fieldOf("value").forGetter(Increment::value),
+            ).apply(instance, Increment::new)
+    );
+
+    public int add(int x) {
+        return value() + x;
+    }
+}
+```
+
+```java
+// Register an enchantment effect component to carry this record.
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ConditionalEffect<Increment>>> INCREMENT =
+    ENCHANTMENT_COMPONENT_TYPES.register("increment",
+        () -> DataComponentType.<ConditionalEffect<Increment>>builder()
+            .persistent(ConditionalEffect.codec(Increment.CODEC, LootContextParamSets.EMPTY))
+            .build());
+```
+
+```java
+// Somewhere in game logic where an `itemStack` is available.
+// `unmodifiedValue` is an integer.
+MutableInt mutableValue = new MutableInt(unmodifiedValue);
+EnchantmentHelper.runIterationOnItem(itemStack, (enchantmentHolder, enchantLevel) -> Enchantment.applyEffects(
+    // Isolates the ConditionalEffect<Increment> from the provided holder and wraps it in a list for applyEffects.
+    enchantmentHolder.value().getEffects(INCREMENT.value()),
```

Comment:
```suggestion
    enchantmentHolder.value().getEffects(INCREMENT.get()),
```

`Holder` would return a `DataComponentType<?>`, which means it would not line up with the generic, causing a compile error unless cast.